bam_data_store/metadata/definitions.py

```diff
@@ -0,0 +1,428 @@
+        """,
+    )
+
+    @model_validator(mode='after')
+    @classmethod
+    def model_validator_after_init(cls, data: Any) -> Any:
+        """
+        Validate the model after instantiation of the class.
+
+        Args:
+            data (Any): The data containing the fields values to validate.
+
+        Returns:
+            Any: The data with the validated fields.
+        """
+        # If `generated_code_prefix` is not set, use the first 3 characters of `code`
+        if not data.generated_code_prefix:
+            data.generated_code_prefix = data.code[:3]
+
+        return data
+
+
+class PropertyTypeDef(EntityDef):
+    """
+    Definition class for a property type. It adds the fields of `property_label`, `data_type`,
+    `vocabulary_code`, `metadata`, `dynamic_script`, and `multivalued` to the common attributes of
+    an entity definition. E.g.:
+
+    ```python
+    class Alias(PropertyTypeDef):
+        defs = PropertyTypeDef(
+            version=1,
+            code='ALIAS',
+            description='
+            e.g. abbreviation or nickname//z.B. Abkürzung oder Spitzname
+            ',
+            data_type='VARCHAR',
+            property_label='Alternative name',
+        )
+    ```
+    """
+
+    property_label: str = Field(
+        ...,
+        description="""
+        Label that appears in the inventory view. This is the human-readable text for the property
+        type definition, and it typically coincides with the `code`, e.g., `'Monitoring date'` for the
+        `MONITORING_DATE` property type.
+        """,
+    )
+
+    data_type: DataType = Field(
+        ...,
+        description="""
+        The data type of the property, i.e., if it is an integer, float, string, etc. The allowed
+        data types in openBIS are:
+            - `BOOLEAN`
+            - `CONTROLLEDVOCABULARY`
+            - `DATE`
+            - `HYPERLINK`
+            - `INTEGER`
+            - `MATERIAL`
+            - `MULTILINE_VARCHAR`
+            - `OBJECT`
+            - `REAL`
+            - `TIMESTAMP`
+            - `VARCHAR`
+            - `XML`
+
+        These are defined as an enumeration in the `DataType` class.
+
+        Read more in https://openbis.readthedocs.io/en/latest/uncategorized/register-master-data-via-the-admin-interface.html#data-types-available-in-openbis.
+        """,
+    )
+
+    vocabulary_code: Optional[str] = Field(
+        default=None,
+        description="""
+        String identifying the controlled vocabulary used for the data type of the property. This is
+        thus only relevant if `data_type == 'CONTROLLEDVOCABULARY'`.
+        """,
+    )
+
+    # TODO add descriptions for `metadata`, `dynamic_script`, and `multivalued`
+
+    metadata: Optional[dict] = Field(
+        default=None,
+        description="""""",
+    )
+
+    dynamic_script: Optional[str] = Field(
+        default=None,
+        description="""""",
+    )
+
+    multivalued: Optional[str] = Field(
```

Comment:
Check if this is being used. For what I know, properties will have just one unique value.

Reply:
Yeah, I don't see anywhere why I added this.

Reply:
I deleted it